### DISCUSSION TITLE
[dev-experience] Strange TS errors in the sources

### DISCUSSION THREAD

**Discussion by kittaakos** [eclipse-theia/theia#7986] [General]
I have seen these kinds of errors in the source (VS Code and Gitpod too) lately when I was working in Theia. It’s not a problem for the build or the runtime, but these errors always show up in the editor. Does anyone know what is causing it? A few examples:

 - `packages/preferences/src/browser/preferences-frontend-application-contribution.ts`:
```
Argument of type 'import("/Users/akos.kitta/Desktop/theia/packages/core/src/common/uri").default' is not assignable to parameter of type 'import("/Users/akos.kitta/Desktop/theia/packages/core/lib/common/uri").default'.
  Types have separate declarations of a private property 'codeUri'.
```

 - `packages/monaco/src/browser/monaco-text-model-service.ts`:
```
Argument of type 'import("/Users/akos.kitta/Desktop/theia/packages/core/src/common/uri").default' is not assignable to parameter of type 'import("/Users/akos.kitta/Desktop/theia/packages/core/lib/common/uri").default'.ts(2345)
```
```
Argument of type 'import("/Users/akos.kitta/Desktop/theia/packages/core/lib/common/resource").Resource' is not assignable to parameter of type 'import("/Users/akos.kitta/Desktop/theia/packages/core/src/common/resource").Resource'.
  Types of property 'uri' are incompatible.
    Type 'import("/Users/akos.kitta/Desktop/theia/packages/core/lib/common/uri").default' is not assignable to type 'import("/Users/akos.kitta/Desktop/theia/packages/core/src/common/uri").default'.
      Types have separate declarations of a private property 'codeUri'.
```

![Screen Shot 2020-06-10 at 11 17 42](https://user-images.githubusercontent.com/1405703/84250200-04666c80-ab0c-11ea-8c1c-38bcefd92238.jpg)

**Comment by akosyakov** [comment]
It looks like latest typescript version does not generate proper types for private properties, so here   `URI.codeUri` type in *.d.ts file does not much to type derived from *.ts file. Maybe there is a way to enable type generation even for private properties as it was before. It is beneficial for downstream when they want to work around private modifiers, besides fixing annoying error.
